src/coffea/dataset_tools/rucio_utils.py

```diff
@@ -0,0 +1,294 @@
+import getpass
+import json
+import os
+import re
+import subprocess
+from collections import defaultdict
+
+from rucio.client import Client
+
+# Rucio needs the default configuration --> taken from CMS cvmfs defaults
+os.environ["RUCIO_HOME"] = "/cvmfs/cms.cern.ch/rucio/current"
+
+
+def get_proxy_path() -> str:
+    """
+    Checks if the VOMS proxy exists and if it is valid
+    for at least 1 hour.
+    If it exists, returns the path of it"""
+    try:
+        subprocess.run("voms-proxy-info -exists -hours 1", shell=True, check=True)
+    except subprocess.CalledProcessError:
+        raise Exception(
+            "VOMS proxy expirend or non-existing: please run `voms-proxy-init -voms cms -rfc --valid 168:0`"
+        )
+
+    # Now get the path of the certificate
+    proxy = subprocess.check_output(
+        "voms-proxy-info -path", shell=True, text=True
+    ).strip()
+    return proxy
+
+
+def get_rucio_client(proxy=None) -> Client:
+    """
+    Open a client to the CMS rucio server using x509 proxy.
+
+    Parameters
+    ----------
+        proxy : str, optional
+            Use the provided proxy file if given, if not use `voms-proxy-info` to get the current active one.
+
+    Returns
+    -------
+        nativeClient: rucio.Client
+            Rucio client
+    """
+    try:
+        if not proxy:
+            proxy = get_proxy_path()
+
+        nativeClient = Client(
+            rucio_host="https://cms-rucio.cern.ch",
```

Comment:
This should automatically be taken from `rucio.cfg` in `$RUCIO_HOME`

Reply:
@valsdav

Reply:
Sorry I wasn't clear here, the point is that the rucio config should provide all necessary arguments and it is sufficient to call `nativeClient = Client()`. e.g.
```
[ncsmith@lxplus909 ~]$ RUCIO_HOME=/cvmfs/cms.cern.ch/rucio/current python3
Python 3.9.18 (main, Sep  7 2023, 00:00:00)
[GCC 11.4.1 20230605 (Red Hat 11.4.1-2)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from rucio.client import Client
>>> client = Client()
>>> client.host
'http://cms-rucio.cern.ch'
>>> client.whoami()
{'email': 'nick.smith@cern.ch', 'deleted_at': None, 'updated_at': '2020-04-28T23:49:13', 'suspended_at': None, 'account_type': 'USER', 'account': 'ncsmith', 'status': 'ACTIVE', 'created_at': '2020-04-28T23:49:13'}
>>>
```

Reply:
I see :) thanks!